### PR TITLE
Cee/offline report

### PR DESCRIPTION
**Pull request by calebeby**
I think this works yall should check

### CODE REVIEW

File: src/api.tsx [hunk 1]
```diff
@@ -19,6 +36,11 @@ const queryAPI = (
     headers: hasValidJWT()
       ? new Headers({ Authentication: `Bearer ${getJWT()}` })
       : undefined
+  }).catch((err: Error) => {
+    if (method !== 'GET') {
+      return addRequestToIdb({ path, method, body })
+    }
+    throw err
```
Comment (GreenBeard) [inline src/api.tsx]: Again, `else`.

File: src/api.tsx [hunk 2]
```diff
@@ -3,11 +3,28 @@ import Match from './models/match'
...
+}
+
+const addRequestToIdb = async (request: req) => {
+  const currentRequests = (await idbKeyval.get('cachedRequests')) as
+    | req[]
+    | undefined
+  if (currentRequests === undefined) {
+    return idbKeyval.set('cachedRequests', [request])
+  }
+  return idbKeyval.set('cachedRequests', currentRequests.concat(request))
```
Comment (GreenBeard) [inline src/api.tsx]: Wouldn't `else` make it more clear?

File: src/index.tsx [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+
+if (navigator.onLine) {
+  idbKeyval
+    .get('cachedRequests')
+    .then((requests: req[]) =>
+      Promise.all(requests.map(re => queryAPI(re.path, re.method, re.body)))
+    )
+    .then(() => idbKeyval.set('cachedRequests', []))
+    .then(() => {
+      console.log('synced them all')
```
Comment (GreenBeard) [inline src/index.tsx]: Either remove the console.log or make it more descriptive.

File: src/api.tsx (hunk 1 above)
Comment (fharding1) [reply src/api.tsx]: No

File: src/api.tsx (hunk 2 above)
Comment (fharding1) [reply src/api.tsx]: No

File: src/index.tsx (hunk 3 above)
Comment (fharding1) [reply src/index.tsx]: :+1: agree

**Comment by GreenBeard** [comment]
It seems to work fine from my testing. However, I don't know how to delete my two fact reports for team 238 :laughing:

File: src/api.tsx (hunk 1 above)
Comment (GreenBeard) [reply src/api.tsx]: For some reason github put this before my other comment about `else`

File: src/api.tsx (hunk 2 above)
Comment (fharding1) [reply src/api.tsx]: AFAIK early returns are favored over if-else in most languages (including JS).

Comment (fharding1) [reply src/api.tsx]: Less cyclomatic complexity is almost always better if possible.

Comment (GreenBeard) [reply src/api.tsx]: Fine I'll take the early return argument. However, it does not introduce any more cyclomatic complexity. There is still a branch at the if statement either way with two possiblities.

Comment (fharding1) [reply src/api.tsx]: True.

**Pull request merged**